noxfile.py

```diff
@@ -0,0 +1,261 @@
+"""
+Nox is a command-line tool that automates testing in multiple Python environments, similar to tox.
+Unlike tox, Nox uses a standard Python file for configuration.
+
+To run all sessions defined in the noxfile.py file, run the following command:
+$ nox
+
+To run a specific session, run the following command:
+$ nox -s <session-name>
+
+To run a session with additional arguments, run the following command:
+$ nox -s <session-name> -- <additional-arguments>
+
+To list all available sessions, run the following command:
+$ nox -l
+"""
+
+import nox
+from nox.sessions import Session
+
+# default sessions (sorted alphabetically)
+nox.options.sessions = ["bandit", "black", "flake8", "isort", "mypy", "unit-tests"]
+
+
+def _install_requirements(session: Session) -> None:
+    session.install("-r", "requirements-extras.txt")
+
+
+@nox.session(reuse_venv=True)
+def bandit(session: Session) -> None:
+    """Run bandit on cachi2 directory"""
+    _install_requirements(session)
+    session.run(
+        "bandit",
+        "-c",
+        "pyproject.toml",
+        "-r",
+        "cachi2",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def black(session: Session) -> None:
+    """Run black on cachi2 and tests directories and noxfile.py"""
+    _install_requirements(session)
+    session.run(
+        "black",
+        "--check",
+        "--diff",
+        "cachi2",
+        "tests",
+        "noxfile.py",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def flake8(session: Session) -> None:
+    """Run flake8 on cachi2 and tests directories"""
+    _install_requirements(session)
+    # TODO: Move flake8-pyproject to requirements-extras.txt
```

Comment:
What's the blocker for this TODO work? Seems simple enough which means I'm missing something.

Reply:
I will do that in the next revision.